email/parse.c

```diff
@@ -1190,21 +1199,23 @@ struct Envelope *mutt_rfc822_read_header(FILE *fp, struct Email *e, bool user_hd
 
   while ((loc = ftello(fp)) != -1)
   {
-    line = mutt_rfc822_read_line(fp, line, &linelen);
-    if (*line == '\0')
+    if ((mutt_rfc822_read_line(fp, line) == 0) || (mutt_buffer_len(line) == 0))
```

Comment:
I don't understand when mutt_buffer_len(line) could be 0 but we could have read more bytes; could that ever happen? If we do, don't we need to seek back?